testing/ef-tests/src/cases/blockchain_test.rs

```diff
@@ -225,11 +232,13 @@ pub fn run_case(
         // Insert the block into the database
         provider
             .insert_block(block.clone(), StorageLocation::Database)
-            .map_err(|err| Error::block_failed(block_number, err))?;
+            .map_err(|err| Error::block_failed(block_number, Default::default(), err))?;
 
         // Consensus checks before block execution
-        pre_execution_checks(chain_spec.clone(), &parent, block)
-            .map_err(|err| Error::block_failed(block_number, err))?;
+        pre_execution_checks(chain_spec.clone(), &parent, block).map_err(|err| {
+            program_inputs.push((block.clone(), execution_witness_with_parent(&parent)));
+            Error::block_failed(block_number, program_inputs.clone(), err)
```

Comment:
@kevaundray, how does this look like (+L456 helper)?

Reply:
LGTM!